packages/type-utils/src/isUnsafeAssignment.ts

```diff
@@ -20,6 +20,7 @@ export function isUnsafeAssignment(
   receiver: ts.Type,
   checker: ts.TypeChecker,
   senderNode: TSESTree.Node | null,
+  visited = new Map<ts.Type, Set<ts.Type>>(),
 ): false | { sender: ts.Type; receiver: ts.Type } {
```

Comment:
[Refactor] This code works great as-is, but has a couple of little paper cuts IMO:

* Each call now has to check whether `visited` exists _(this is not really a big issue at all, certainly not a blocker)_
* Callers now can see this `visited` `Map` even though there's no reason for them to provide it

What do you think about making a non-exported worker function so the public signature doesn't need to have `visited`?

```ts
  senderNode: TSESTree.Node | null,
): false | { sender: ts.Type; receiver: ts.Type } {
  return isUnsafeAssignmentWorker(type, receiver, checker, senderNode, new Map());
```

Reply:
> What do you think about making a non-exported worker function so the public signature doesn't need to have visited?

I think it's much better. let me refactor it! :)